RFC-0004-sparse-fill-value/README.md

```diff
@@ -0,0 +1,613 @@
+also need to consider:
+- how the nonzero fill value affects the result of linear algebra
+  operations such as matrix multiplication, etc.
+- how to define a fill value for hybrid tensors (tensors with sparse
+  and dense dimensions),
+- how PyTorch autograd support needs to deal with a specified
+  fill value.
+
+
+## Proposal
+
+This proposal is about enabling sparse tensor support for the Calculus
+domain while preserving the existing functionality for the Linear
+Algebra domain, and allowing extensions to the Graph domain.
+
+In the following, we describe the sparse tensor `fill_value` feature
+with examples involving sparse tensors in COO storage format.  When a
+new sparse tensor storage format is introduced to PyTorch, the same
+semantics should be applied to the new format.
+
+0.  Used terminology
+
+    - "Defined value" is a value for which memory is allocated and
+      initialized.
+    - "Undefined value" is a value for which memory is allocated, but
+      the content of the memory can be arbitrary.
+    - "Indefinite value" represents a structural lack of value.
+    - "Sparse tensor format" is a memory-efficient storage format for tensors
+      with many equal elements.
+    - "Strided tensor format" is a process-efficient storage format
+      for general tensors.
+    - "Dense tensor" is a tensor with many non-equal elements. Dense
+      tensors can be stored in any tensor format, including a sparse
+      tensor format.
+    - "Dense part" is a strided tensor that is obtained by fixing the
+      indices of all sparse dimensions in a hybrid tensor.
+
+1.  We propose to extend sparse tensor constructors with a keyword
+    argument `fill_value`, used to define the value for
+    unspecified elements of the constructed sparse tensor.
+
+    For instance, the Python signature of `sparse_coo_tensor` would be:
+
+    ```python
+    torch.sparse_coo_tensor(indices, values, size=None, fill_value=None, dtype=None, device=None, requires_grad=False)
+    ```
+
+    where `fill_value=None` indicates the constructor to use the
+    default fill value.
+
+2.  The default fill value is zero.
+
+    This choice is consistent with the interpretation of an unspecified
+    element in Linear Algebra.
+
+    The default fill value of a sparse non-hybrid tensor is a scalar
+    tensor: `torch.tensor(0, dtype=A.dtype, device=A.device)`. For the
+    default fill value of a hybrid tensor, see points 5 and 6 below.
+
+3.  PyTorch functions that have the `layout` argument may use a different
+    `fill_value` when constructing a sparse tensor as defined by the
+    following table:
+
+    | Function              | `fill_value` of returned sparse tensor |
+    | :-------------------- | :----------------------------------- |
+    | `torch.empty`         | undefined value                      |
+    | `torch.empty_like`    | undefined value                      |
+    | `torch.empty_strided` | N/A                                  |
```

Comment:
This function makes no sense for sparse tensors, it makes no sense to include it in this table.
```suggestion
```

Reply:
ok